src/WakuRlnV2.sol

```diff
@@ -230,33 +222,56 @@ contract WakuRlnV2 is Initializable, Ownable2StepUpgradeable, UUPSUpgradeable, M
         return fixedSizeProof;
     }
 
-    /// @notice Extend a grace-period membership
+    /// @notice Extend a grace-period membership under the same conditions
     /// @param idCommitments list of idCommitments of memberships to extend
     function extendMemberships(uint256[] calldata idCommitments) external {
         for (uint256 i = 0; i < idCommitments.length; i++) {
-            uint256 idCommitmentToExtend = idCommitments[i];
-            _extendMembership(_msgSender(), idCommitmentToExtend);
+            _extendMembership(_msgSender(), idCommitments[i]);
         }
     }
 
     /// @notice Erase expired memberships or owned grace-period memberships
-    /// The user can select expired memberships offchain,
-    /// and proceed to erase them.
-    /// This function is also used to erase the user's own grace-period memberships.
-    /// The user (i.e. the transaction sender) can then withdraw the deposited tokens.
-    /// @param idCommitments list of idCommitments of the memberships to erase
+    /// The user can select expired memberships offchain, and proceed to erase them.
+    /// The holder can use this function to erase their own grace-period memberships.
+    /// The holder can then withdraw the deposited tokens.
+    /// @param idCommitments The list of idCommitments of the memberships to erase
+    /// set
     function eraseMemberships(uint256[] calldata idCommitments) external {
-        for (uint256 i = 0; i < idCommitments.length; i++) {
-            // FIXME: not DRY: see register()
-            uint256 idCommitmentToErase = idCommitments[i];
-            MembershipInfo memory membershipToErase = memberships[idCommitmentToErase];
-            if (membershipToErase.rateLimit == 0) revert InvalidIdCommitment(idCommitmentToErase);
-            _eraseMembershipAndSaveSlotToReuse(_msgSender(), idCommitmentToErase, membershipToErase);
-            LazyIMT.update(merkleTree, 0, membershipToErase.index);
+        _eraseMemberships(idCommitments, false);
+    }
+
+    /// @notice Erase expired memberships or owned grace-period memberships
+    /// Optionally, also erase rate commitment data from the membership set (clean-up).
+    /// Compared to eraseMemberships(idCommitments),
+    /// this fucntion decreases Merkle tree size and spends more gas (if eraseFromMembershipSet == true).
+    /// @param idCommitments The list of idCommitments of the memberships to erase
+    /// @param eraseFromMembershipSet Indicates whether to erase membership data from the membership set
+    function eraseMemberships(uint256[] calldata idCommitments, bool eraseFromMembershipSet) external {
```

Comment:
I guess setting `eraseFromMembershipSet` to `true` could be taken into account in the future to bring additional rewards to whoever call the function if we decide to choose that route, right? otherwise there's no incentive to pass `true` since it will cost extra gas with no benefit.

Reply:
My idea here is that we don't want to burden users with extra gas cost for cleaning up the tree, but "we" (as contract administrators) may call `eraseMemberships` with `eraseFromMembershipSet = true` when necessary to clean up the tree at our own expense. I agree that we could plug in a reward mechanism for third parties to do this clean-up if we deem it necessary.